tests/sync.rs

```diff
@@ -1,4 +1,4 @@
-use super::*;
+use {super::*, tokio_util::sync::CancellationToken};
 
 pub(crate) static BATCH_COUNTER: AtomicUsize = AtomicUsize::new(0);
```

Comment:
If we batch by ID instead of timestamp it will reduce collision risk, eliminate manual state management, increase thread-safety and guarantees determinism. Just make sure you replace references to
`BATCH_COUNTER.fetch_add(1, Ordering::SeqCst) as u64` with `next_test_batch_id()` if you decide to do this.
```suggestion
pub(crate) fn next_test_batch_id() -> u64 {
    use std::sync::atomic::{AtomicU64, Ordering};
    static COUNTER: AtomicU64 = AtomicU64::new(0);
    COUNTER.fetch_add(1, Ordering::Relaxed)
}
```